From 4bb95bea3b7ff2ae5aca4ee9d56f6c1af20c62e9 Mon Sep 17 00:00:00 2001
From: James Zern <jzern@google.com>
Date: Tue, 11 Jun 2024 13:55:36 -0700
Subject: [PATCH 4/4] configure: add -c to ASFLAGS for android + AS=clang

The GNU Assembler was removed in r24. clang's internal assembler works,
but `-c` is necessary to avoid linking.

Bug: webm:1856
Change-Id: I61f80cf78657d3b71d5e73c5b2510575533ca5ea
---
 build/make/configure.sh | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 629d79f53..c4524b7a8 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1123,6 +1123,14 @@ EOF
           echo "See build/make/Android.mk for details."
           check_add_ldflags -static
           soft_enable unit_tests
+          case "$AS" in
+            *clang)
+              # The GNU Assembler was removed in the r24 version of the NDK.
+              # clang's internal assembler works, but `-c` is necessary to
+              # avoid linking.
+              add_asflags -c
+              ;;
+          esac
           ;;
 
         darwin)
-- 
2.43.0

